texts/0006-tauri-releases-schedule.md

```diff
@@ -0,0 +1,55 @@
+- Feature Name: tauri_release_schedule
+- Start Date: 2024-6-6
+- RFC PR: [tauri-apps/rfcs#14](https://github.com/tauri-apps/rfcs/pull/14)
+- Tracking Issue:
+
+# Summary
+
+Define a release schedule for tauri main repo, including crates, NPM packages and CLIs. 
+
+# Motivation
+
+As it stands now, there is no release schedule and releases are made either whenever one of the working group feels like it (mainly Lucas and myself (Amr)),
+or when there is a need for an urgent fix.
+
+As a tauri user or contributor, there is no indication when a release might happen and if will it include a certain feature
+one might be waiting for which leads me to either try to figure how to use unreleased features directly from source or wait indefinitely.   
+
+A release schedule should make the releasing process more predictable for maintainers and users and prevents the tauri project from ever appearing stagnant.
+
+# Explanation
+
+We follow semver versioning for all tauri crates, NPM packages and CLIs and so we have 6 type of releases we are ever gonna make for stable releases,
+here is what each release type cycle would look like:
+
+| patch *   | minor     | major  | alpha  | beta     | rc     *   |
+|---------------|---------|-----------|--------|--------|----------|
+| monthly | bimonthly | yearly | 2 each week or as needed | weekly | as needed |
+| 12~ per year | 6 per year | 1 per year | 12~ in 1 and a half months | 4~ in one month | 4~ in one month |
```

Comment:
For `patch`/`alpha`/`beta`/`rc`, I'd suggest specifying targets as minimums:

```suggestion
- `patch`: At least monthly
- `minor`: Every 2 months
- `major`: Yearly
- `alpha`: At least 2 per week
- `beta`: At least weekly
- `rc`: At least weekly
```

Reply:
Minimums seems more enforcing, and I don't want to enforce a release if there is nothing to release.

Reply:
For patches i'd like to have something like "If there are patch changes, release them every (second) Monday (unless urgent)". We often get feedback that patches are released too slow so imo we should really just constantly release new ones... This may require 2 branches though so we can still merge minors.

I think that a yearly major is way too ambitious (and too expensive lol). Considering their breaking nature i'd lean towards requiring a 1 year gap between majors with no upper bound (meaning 3 years would be fine too if we have no need for it). idk how realistic that is looking at win7 and webkitgtk (at least for v3).

Reply:
I don't mind patches every week as well

Reply:
I don't mind patches every week as well, we could have a policy to prevent merging PRs that will bump `minor` and just mark them as "ready or pending for next minor" and on the next minor release date, we merge those pending PRs and release.

Reply:
I have updated the schedule as per this discussion, please review and let me know if there is more to adjust.

Reply:
Even though it's not relevant anymore since it's been clarified, it's still worth noting that "bimonthly" can be ambiguous, as it can mean either "twice per month" or "every two months"